src/slurm_plugin/clustermgtd.py

```diff
@@ -543,91 +521,54 @@ def _get_ec2_instances(self):
                     health_check_type=ClusterManager.HealthCheckTypes.ec2_health,
                 )
             # Perform scheduled event actions
             if not self._config.disable_scheduled_event_health_check:
                 self._handle_health_check(
-                    unhealthy_instance_status,
-                    id_to_instance_map,
-                    ip_to_slurm_node_map,
+                    unhealthy_instances_status,
+                    instance_id_to_active_node_map,
                     health_check_type=ClusterManager.HealthCheckTypes.scheduled_event,
                 )
 
-    @staticmethod
-    def _fail_ec2_health_check(instance_health_state, current_time, health_check_timeout):
-        """Check if instance is failing any EC2 health check for more than health_check_timeout."""
-        try:
-            if (
-                # Check instance status
-                instance_health_state.instance_status.get("Status") in EC2_HEALTH_STATUS_UNHEALTHY_STATES
-                and time_is_up(
-                    instance_health_state.instance_status.get("Details")[0].get("ImpairedSince"),
-                    current_time,
-                    health_check_timeout,
-                )
-            ) or (
-                # Check system status
-                instance_health_state.system_status.get("Status") in EC2_HEALTH_STATUS_UNHEALTHY_STATES
-                and time_is_up(
-                    instance_health_state.system_status.get("Details")[0].get("ImpairedSince"),
-                    current_time,
-                    health_check_timeout,
-                )
-            ):
-                return True
-        except Exception as e:
-            log.warning("Error when parsing instance health status %s, with exception: %s", instance_health_state, e)
-            return False
-
-        return False
-
-    @staticmethod
-    def _fail_scheduled_events_check(instance_health_state):
-        """Check if instance has EC2 scheduled maintenance event."""
-        if instance_health_state.scheduled_events:
-            return True
-        return False
-
-    @log_exception(log, "handling health check", catch_exception=Exception, raise_on_error=False)
-    def _handle_health_check(
-        self, unhealthy_instance_status, id_to_instance_map, ip_to_slurm_node_map, health_check_type
+    def _get_nodes_failing_health_check(
+        self, unhealthy_instances_status, instance_id_to_active_node_map, health_check_type
     ):
-        """
-        Perform supported health checks action.
-
-        Place nodes failing health check into DRAIN, so they can be maintained when possible
-        """
+        """Get nodes fail health check."""
         health_check_functions = {
             ClusterManager.HealthCheckTypes.scheduled_event: {
-                "function": ClusterManager._fail_scheduled_events_check,
+                "function": EC2InstanceHealthState.fail_scheduled_events_check,
```

Comment:
I don't think this looks correct. `fail_scheduled_events_check` is not a static method and is bound with an instance of a class(i.e. `self`). If we want to use it like this we need to declare this function as a static method.

Reply:
The origin code have self inside the function, which is correct but may be hard to maintain. After discuss with Mauri, add another function to make it more clear:
```
def _is_instance_unhealthy(self, instance_status: EC2InstanceHealthState, health_check_type):
        """Check if instance status is unhealthy based on picked corresponding health check function."""
        is_instance_status_unhealthy = False
        if health_check_type is ClusterManager.HealthCheckTypes.scheduled_event:
            is_instance_status_unhealthy = instance_status.fail_scheduled_events_check()
        elif health_check_type is ClusterManager.HealthCheckTypes.ec2_health:
            is_instance_status_unhealthy = instance_status.fail_ec2_health_check(
                self._current_time, self._config.health_check_timeout
            )
        return is_instance_status_unhealthy
```